server/controllers/chartsController/helpers.js

```diff
@@ -303,3 +333,95 @@ export const calculateSubjectVariableDayCount = (
 
   return subjectsDayDataAndVariable.length
 }
+
+export const buildFacetForFilters = ({
+  isSexAtBirthFilterActive,
+  isInclusionCriteriaFilterActive,
+  userAccess,
+}) => {
+  const facetForFilters = {}
+
+  if (isSexAtBirthFilterActive) {
+    facetForFilters.socioDemographics = [
+      {
+        $match: {
+          assessment: SOCIODEMOGRAPHICS_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+        },
+      },
+      {
+        $addFields: {
+          filter: SOCIODEMOGRAPHICS_FORM,
+        },
+      },
+    ]
+  }
+  if (isInclusionCriteriaFilterActive) {
+    facetForFilters.inclusionCriteria = [
+      {
+        $match: {
+          assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+        },
+      },
+      {
+        $addFields: {
+          filter: INCLUSION_EXCLUSION_CRITERIA_FORM,
+        },
+      },
+    ]
+  }
+
+  return facetForFilters
+}
+
+export const intersectSubjectsFromFilters = (filters) => {
+  const intersectedSubjects = new Map()
+  const [filterListA, ...rest] = Object.values(filters).map((filter) => {
+    const filterMap = new Map()
+    filter.forEach((subjectData) => {
+      filterMap.set(subjectData.subject, { ...subjectData })
+    })
+
+    return filterMap
+  })
+  filterListA.forEach((subjectData) => {
+    const { subject, collection, filter } = subjectData
+    intersectedSubjects.set(subject, {
+      subject,
+      collections: [{ collection, filter }],
+    })
```

Comment:
since you already have the subject as the key you could make the value just the array of collections

```js
intersectedSubjects.set(subject, [{ collection, filter }])
```

then below when you add to the map you could do

```js
intersectedSubjects.set(subjectData.subject, existingSubjectData.concat([
  { collection: filterSubjectData.collection, filter: filterSubjectData.filter },
]))
```

and then you would use the key when you iterate through this map

```diff
  await Promise.all(
    Array.from(intersectedSubjectsFromFilters).map(
-     async ([_, { subject, collections }]) => {
+     async ([subject, collections]) => {
        // rest of the logic
      })
    )
  )
```

This is a bit of a nit because it'll work either way but I think this simplifies the logic and data structure.